src/append/console.rs

```diff
@@ -181,7 +185,27 @@ impl ConsoleAppenderBuilder {
     }
 
     /// Consumes the `ConsoleAppenderBuilder`, producing a `ConsoleAppender`.
+    #[cfg(feature = "pattern_encoder")]
     pub fn build(self) -> ConsoleAppender {
+        self.build_internal(
+            |this| {
+                this.encoder.take().unwrap_or_else(|| Box::new(PatternEncoder::default()))
+            }
+        )
+    }
+
+    /// Consumes the `ConsoleAppenderBuilder`, producing a `ConsoleAppender`.
+    #[cfg(not(feature = "pattern_encoder"))]
+    pub fn build(self, encoder: Box<dyn Encode>) -> ConsoleAppender {
+        self.build_internal(|_| encoder)
+    }
+
+    fn build_internal<F>(mut self, encoder: F) -> ConsoleAppender
+    where
+        F: FnOnce(&mut Self) -> Box<dyn Encode>,
```

Comment:
Should be FnOnce(Self) if we're consuming right?

Reply:
That would cause issues since other fields need to be moved out of after that function is called. To make that function align better with what's semantically happening, it could perhaps be rewritten to perform an initial destructuring step that splats out the builder into its fields and deals with converting out of the option.

Reply:
If you think that would make more sense then I can take care of that later today when I also resolve the CI issues